Library/ViewModels/DashboardViewModel.swift

```diff
@@ -32,6 +32,12 @@ public protocol DashboardViewModelInputs {
   /// Call when the projects drawer has animated out.
   func dashboardProjectsDrawerDidAnimateOut()
 
+  /// Call to open message thread for specific project
+  func messageThreadNavigated(projectId: Param, messageThread: MessageThread)
+
+  /// Call to open project messages thread
+  func openMessageThreadRequested()
```

Comment:
I think that the cell being tapped is the event that we're responding to here. Perhaps there is another alternative name, I just think that _open_ and _requested_ make this seem like an output rather than an input.

Also if there were another way of triggering this we would represent that as a separate input so being specific about exactly what the user did is fine I would say.